Restart ssh after updating host keys.

nuclearsandwich opened the pull request:
I did this completely blind and haven't yet tested it at all.

nuclearsandwich committed:
Restart ssh after updating host keys.

Crola1702 committed:
Fix undefined method notifies without resources

Repo recipe run is showing:
* NoMethodError: undefined method `notifies' for cookbook: ros_buildfarm"

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

Crola1702 commented:
Just added c587ffd. `notifies` was outside a resource block.

I tested it and this change shows at the end of the chef run:
```
  * service[ssh] action restart
    - restart service service[ssh]
```

nuclearsandwich committed:
Notify in each file rather than at the end of a loop.

This was the intent and operating in the context of a file resource will
mean that sshd only restarts when one of these files actually changes.
Which is preferred rather than notifying unconditionally as it would
with the ruby_block resource.
